### ISSUE TITLE
Webstorm and sabarivka fails to work together when code coverage needed. Works fine from ng test

### ISSUE THREAD

**Issue by criskrzysiu** [kopach/karma-sabarivka-reporter#254]
karma.config.js
```js
// Karma configuration file, see link for more information
// https://karma-runner.github.io/1.0/config/configuration-file.html

module.exports = function (config) {
  process.on('infrastructure_error', (error) => {
    console.error('infrastructure_error', error);
  });

  config.set({
    basePath: '',
    frameworks: ['jasmine', '@angular-devkit/build-angular'],
    plugins: [
      require('karma-jasmine'),
      require('karma-chrome-launcher'),
      require('karma-sabarivka-reporter'),
      require('karma-jasmine-html-reporter'),
      require('karma-coverage-istanbul-reporter'),
      require('@angular-devkit/build-angular/plugins/karma'),
      require('karma-junit-reporter'),
      'karma-spec-reporter'
    ],
    client: {
      clearContext: false // leave Jasmine Spec Runner output visible in browser
    },
    coverageIstanbulReporter: {
      dir: require('path').join(__dirname, './coverage/project'),
      reports: ['cobertura', 'lcovonly', 'html'],
      fixWebpackSourcePaths: true,
      thresholds: {
        emitWarning: true, // set to `true` to not fail the test command when thresholds are not met
        // thresholds for all files
        global: {
          statements: 90,
          lines: 90,
          branches: 90,
          functions: 90
        }
      }
    },
    reporters: ['sabarivka', 'progress', 'kjhtml', 'junit'],
    coverageReporter: {
      include: ['src/app/**/*.(ts|js)', '!src/**/(*.spec|main|*module|environment.*).(ts|js)']
    },
    junitReporter: {
      outputDir: '', // results will be saved as $outputDir/$browserName.xml
      outputFile: undefined, // if included, results will be saved as $outputDir/$browserName/$outputFile
      suite: '', // suite will become the package name attribute in xml testsuite element
      useBrowserName: true, // add browser name to report and classes names
      nameFormatter: undefined, // function (browser, result) to customize the name attribute in xml testcase element
      classNameFormatter: undefined, // function (browser, result) to customize the classname attribute in xml testcase element
      properties: {}, // key value pair of properties to add to the <properties> section of the report
      xmlVersion: null // use '1' if reporting to be per SonarQube 6.2 XML format
    },
    port: 9876,
    colors: true,
    logLevel: config.LOG_INFO,
    autoWatch: true,
    browsers: ['Chrome'],
    singleRun: false,
    restartOnFileChange: true,
    verbose: true
  });
};
```

```
[
  'sabarivka',
  'kjhtml',
  'junit',
  'intellij_c831a91b03572bad3b3db88354641e3b'
]
{"LOG_DISABLE":"OFF","LOG_ERROR":"ERROR","LOG_WARN":"WARN","LOG_INFO":"INFO","LOG_DEBUG":"DEBUG","frameworks":["jasmine","@angular-devkit/build-angular"],"protocol":"http:","port":9876,"listenAddress":null,"hostname":null,"httpsServerConfig":{},"basePath":"%project%","files":[],"browserConsoleLogOptions":{"level":"debug","format":"%b %T: %m","terminal":true},"customContextFile":null,"customDebugFile":null,"customClientContextFile":null,"exclude":["%project%\\karma.conf.js"],"logLevel":"INFO","colors":true,"autoWatch":false,"autoWatchBatchDelay":0,"restartOnFileChange":true,"usePolling":false,"reporters":["sabarivka","kjhtml","junit","intellij_c831a91b03572bad3b3db88354641e3b","intellijCoverage_33e284dac2b015a9da50d767dc3fa58a"],"singleRun":false,"browsers":["Chrome"],"captureTimeout":60000,"pingTimeout":5000,"proxies":{},"proxyValidateSSL":true,"preprocessors":{},"preprocessor_priority":{},"urlRoot":"/","reportSlowerThan":0,"loggers":[{"type":"console","layout":{"type":"pattern","pattern":"%[%d{DATE}:%p [%c]: %]%m"}}],"transports":["polling","websocket"],"forceJSONP":false,"plugins":[{"framework:jasmine":["factory",null],"reporter:karma-jasmine":["factory",null]},{"launcher:Chrome":["type",null],"launcher:ChromeHeadless":["type",null],"launcher:ChromeCanary":["type",null],"launcher:ChromeCanaryHeadless":["type",null],"launcher:Chromium":["type",null],"launcher:ChromiumHeadless":["type",null],"launcher:Dartium":["type",null],"test":{}},{"reporter:karma-sabarivka-reporter":["type",null],"reporter:sabarivka":["type",null]},{"reporter:kjhtml":["type",null]},{"reporter:coverage-istanbul":["type",null]},{"framework:@angular-devkit/build-angular":["factory",null],"reporter:@angular-devkit/build-angular--sourcemap-reporter":["type",null],"reporter:@angular-devkit/build-angular--event-reporter":["type",null],"middleware:@angular-devkit/build-angular--blocker":["factory",null],"middleware:@angular-devkit/build-angular--fallback":["factory",null]},{"reporter:junit":["type",null]},"karma-spec-reporter","%user%\\AppData\\Local\\JetBrains\\Toolbox\\apps\\WebStorm\\ch-0\\202.7319.49\\plugins\\js-karma\\js_reporter\\karma-intellij\\lib\\intellijPlugin.js"],"client":{"args":[],"useIframe":true,"runInParent":false,"captureConsole":true,"clearContext":false},"defaultClient":{"args":[],"useIframe":true,"runInParent":false,"captureConsole":true,"clearContext":false},"browserDisconnectTimeout":2000,"browserDisconnectTolerance":0,"browserNoActivityTimeout":30000,"processKillTimeout":2000,"concurrency":null,"failOnEmptyTestSuite":true,"retryLimit":2,"detached":false,"crossOriginAttribute":true,"browserSocketTimeout":20000,"configFile":"%user%\\AppData\\Local\\JetBrains\\Toolbox\\apps\\WebStorm\\ch-0\\202.7319.49\\plugins\\js-karma\\js_reporter\\karma-intellij\\lib\\intellij.conf.js","coverageIstanbulReporter":{"dir":"%project%\\coverage\\project","reports":["cobertura","lcovonly","html"],"fixWebpackSourcePaths":true,"thresholds":{"emitWarning":true,"global":{"statements":90,"lines":90,"branches":90,"functions":90}}},"coverageReporter":{"include":["src/app/**/*.(ts|js)","!src/**/(*.spec|main|*module|environment.*).(ts|js)"]},"junitReporter":{"outputDir":"","suite":"","useBrowserName":true,"properties":{},"xmlVersion":null},"verbose":true}
30 09 2020 16:05:25.851:INFO [karma-server]: Karma v5.2.3 server started at http://localhost:9876/
30 09 2020 16:05:25.856:INFO [launcher]: Launching browsers Chrome with concurrency unlimited
30 09 2020 16:05:25.860:INFO [launcher]: Starting browser Chrome
30 09 2020 16:06:07.991:INFO [Chrome 85.0.4183.121 (Windows 10)]: Connected on socket lixOGu_8HxtfQQywAAAA with id 78815095
30 09 2020 16:06:08.394:WARN [karma]: Error during refresh file list. TypeError: Cannot read property 'push' of undefined
    at %project%\node_modules\karma\lib\reporters\multi.js:11:61
    at Array.forEach (<anonymous>)
    at MultiReporter.addAdapter (%project%\node_modules\karma\lib\reporters\multi.js:11:21)
    at Server.<anonymous> (%project%\node_modules\karma\lib\middleware\runner.js:97:18)
    at Object.onceWrapper (events.js:422:26)
    at Server.emit (events.js:327:22)
    at Executor.schedule (%project%\node_modules\karma\lib\executor.js:31:20)
    at %project%\node_modules\karma\lib\middleware\runner.js:42:18
30 09 2020 16:06:08.422:ERROR [karma-server]: UncaughtException:: Cannot read property 'write' of null
infrastructure_error TypeError: Cannot read property 'write' of null
    at %user%\AppData\Local\JetBrains\Toolbox\apps\WebStorm\ch-0\202.7319.49\plugins\js-karma\js_reporter\karma-intellij\lib\intellijReporter.js:151:12
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
30 09 2020 16:06:08.627:ERROR [karma-server]: TypeError: Cannot read property 'write' of null
    at %user%\AppData\Local\JetBrains\Toolbox\apps\WebStorm\ch-0\202.7319.49\plugins\js-karma\js_reporter\karma-intellij\lib\intellijReporter.js:151:12
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
30 09 2020 16:06:08.628:ERROR [karma-server]: UncaughtException:: Cannot read property 'write' of null
infrastructure_error TypeError: Cannot read property 'write' of null
    at %user%\AppData\Local\JetBrains\Toolbox\apps\WebStorm\ch-0\202.7319.49\plugins\js-karma\js_reporter\karma-intellij\lib\intellijReporter.js:151:12
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
30 09 2020 16:06:08.629:ERROR [karma-server]: TypeError: Cannot read property 'write' of null
    at %user%\AppData\Local\JetBrains\Toolbox\apps\WebStorm\ch-0\202.7319.49\plugins\js-karma\js_reporter\karma-intellij\lib\intellijReporter.js:151:12
    at processTicksAndRejections (internal/process/task_queues.js:79:11)

Process finished with exit code 1
```

**Comment by criskrzysiu** [comment]
all packages latest (karma/angular/sabarivka)

**Comment by kopach** [comment]
Hi @criskrzysiu. If you take a look on logs – there is no error in `savarivka` reporter, it works just as it should (as any other karma plugin). Instead, your error comes from `js_reporter\karma-intellij`. So, you, probably, need to report this issue to that project.

**Comment by kopach** [comment]
Also, the fact, that everything works fine from CLI is additional proof, that issue not in `karma-sabarivka-reporter` itself.

**Comment by criskrzysiu** [comment]
ok I know whats the problem and it actually should be fixed in this plugin. Intelij's plugin is using causing runing karma multi reporter MultiReporter.addAdapter that looks like this
```ts
  addAdapter (adapter) {
    this._reporters.forEach((reporter) => reporter.adapters.push(adapter))
  }
```
and as this plugin does not have adapters array we get error. 

After adding adapters array everything works fine :)

**Comment by kopach** [comment]
hi @criskrzysiu. Thank you for your work done here. Would you like to rise PR for this fix, as you already have done all the job? :)

**Comment by kopach** [comment]
hi @criskrzysiu. Fix was done and introduced in `karma-sabarivka-reporter@3.2.5`. Please, check it out if it works for you and let me know. Thanks again for your work done here. Please, consider also adding your star to this project if you like it. Thanks.

**Comment by kopach** [comment]
hi @brannislav, thanks for this info. Would you like to rise PR with the fix yourself, maybe?

**Comment by kopach** [comment]
Fix is released in v3.3.0.
